### DISCUSSION TITLE
Best way to use the same map with multiple routes?

### DISCUSSION THREAD

**Discussion by michaelhthomas** [fleaflet/flutter_map#1451] [Q&A]
I'm building an interface akin to that of Google Maps, with several views which each should contain the map widget (home, search, location details, etc.). I'm trying to determine what the best approach will be / if it is possible to use flutter's routing (i'm using `auto_route`) to have each of these views be different "routes". I was originally planning on having each of these routes appear above the map element, which would be in the parent route. Unfortunately I've had a number of unique struggles that make me think this isn't the best option. Has anyone else tried to to something similar, and is there a better way that I'm missing?

**Comment by codeOfJannik** [comment]
Did you find a solution? I have similar issue currently, want to be able to reuse the same map as a hero widget between multiple page routes

**Reply by michaelhthomas** [reply]
I never figured it out and ended up just giving up. Hopefully someone with more flutter experience can figure this one out.
